### PR TITLE
request to merge assignment 5 into develop branch which contains question based on axios, graphql

### PR DESCRIPTION
**Pull request by munendra-successive**
1.Create a React component that fetches data from a public API (e.g., JSONPlaceholder) and displays it on the page. Use the fetch function to make the request.

2.Implement error handling for the fetch request. Display an error message if the request fails and provide a way for the user to retry the request.

3.Add a loading indicator (e.g., a spinner) to the component. Display it while the data is being fetched and hide it once the data is loaded.

4.Fetch a large dataset from an API and implement pagination. Display a limited number of items per page and provide navigation controls to load more pages of data.

5.Create a React component that fetches data from a public API (e.g., JSONPlaceholder) using Axios and displays it on the page.

6.Modify the previous assignment to implement error handling using Axios. Display an error message if the request fails and provide a way for the user to retry the request.

7.Create a form in your React component that allows the user to input data (e.g., a post title and content). Use Axios to send a POST request to an API endpoint, posting the user's data.

8.set up a GraphQL client library like Apollo Client. Create a simple component that uses the client to fetch and display data from a GraphQL API.

9.Create a React component that makes GraphQL queries to retrieve a list of items (e.g., a list of books, movies, or products) from a GraphQL API. Implement pagination for large datasets and display the data in a user-friendly way.

12.Create a reusable Button component in React that accepts props for different styles (e.g., primary, secondary, danger) and click handlers. Demonstrate how to use this component in a sample application.

13.Design and implement a Modal component in React. Allow customization of the modal content and provide options to open, close, and toggle the modal. Use this component to display various types of content.

14.Create an HOC named withAuth that restricts access to a specific component to authenticated users only. Implement this HOC on a sample component and demonstrate how it protects routes.

15.Build an HOC called withLogger that logs the events of a component (e.g., mounting, unmounting, updating). Apply this HOC to a component and display the logs in the browser console.

16.Develop an HOC named withDataFetching that fetches data from an API and passes it as props to a component. Use this HOC to retrieve and display data in a component.

### CODE REVIEW

File: src/assignment-5/QuestionTweleve/ButtonComp.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Button } from "@mui/material";
+
+const ButtonComp = (props) => {
+  const { buttonstyles, handleClick1, handleClick2, handleClick3 } = props.name;
+  return (
```
Comment (UrvashiChaudhary) [inline src/assignment-5/QuestionTweleve/ButtonComp.jsx:5]: remove props.name and use correct destructuring techniques

File: src/assignment-5/QuestionThirteen/ModalComp.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const [openhome, setOpenHome] = useState(false);
+  const [openabout, setOpenAbout] = useState(false);
+  const [opencontact, setOpenContact] = useState(false);
+
+  const handleOpenhome = () => setOpenHome(true);
+  const handleClosehome = () => setOpenHome(false);
+  const handleOpenabout = () => setOpenAbout(true);
+  const handleCloseabout = () => setOpenAbout(false);
+  const handleOpencontact = () => setOpenContact(true);
+  const handleClosecontact = () => setOpenContact(false);
```
Comment (UrvashiChaudhary) [inline src/assignment-5/QuestionThirteen/ModalComp.jsx:39]: Instead of using these handler you can simply set state onClick

**Review by UrvashiChaudhary — CHANGES_REQUESTED** [review]

File: src/assignment-5/QuestionOne.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+
+1.Create a React component that fetches data from a public API (e.g., JSONPlaceholder) 
+and displays it on the page. Use the fetch function to make the request.
+
+
+*/
+
+import { useEffect } from "react";
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionOne.jsx:9]: Done

File: src/assignment-5/QuestionOne.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <>
+      <h4>
+        1.Create a React component that fetches data from a public API (e.g.,
+        JSONPlaceholder) and displays it on the page. Use the fetch function to
+        make the request.
+      </h4>
+      <h1>Data from API</h1>
+      <table>
+        <tr>
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionOne.jsx:30]: Done

File: src/assignment-5/QuestionOne.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          <th>Name</th>
+          <th>Username</th>
+          <th>Email</th>
+        </tr>
+        {data &&
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionOne.jsx:35]: Done

File: src/assignment-5/QuestionOne.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+and displays it on the page. Use the fetch function to make the request.
+
+
+*/
+
+import { useEffect } from "react";
+import { useState } from "react";
+
+const QuestionOne = () => {
+  const [data, setData] = useState(null);
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionOne.jsx:13]: assigned initial state as empty array

File: src/assignment-5/QuestionTwo.jsx
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionTwo.jsx:1]: Done

File: src/assignment-5/QuestionFive.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+import { useEffect } from "react";
+import { useState } from "react";
+/*
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionFive.jsx:4]: done

File: src/assignment-5/QuestionFive.jsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <>
+      <h4>
+        5.Create a React component that fetches data from a public API (e.g.,
+        JSONPlaceholder) using Axios and displays it on the page.
+      </h4>
+      {data ? (
+        data && (
+          <table>
+            <tr>
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-5/QuestionFive.jsx:27]: done

File: src/assignment-5/QuestionTweleve/ButtonComp.jsx (hunk 1 above)
Comment (munendra-successive) [reply src/assignment-5/QuestionTweleve/ButtonComp.jsx:5]: done

**Pull request merged**